djangocms_snippet/admin.py

```diff
@@ -68,5 +69,31 @@ def has_delete_permission(self, request, obj=None):
             )
         return False
 
+    def has_change_permission(self, request, obj=None):
+        return False
+
+    def _get_preview_url(self, obj):
+        """
+        Return the change url which will be rendered in read only mode
+        :return: method or None
+        """
+        change_url = reverse(
+            "admin:{app}_{model}_change".format(
+                app=obj._meta.app_label, model=self.model._meta.model_name
+            ),
+            args=(obj.pk,),
+        )
+        return change_url
+
+    def get_list_actions(self):
```

Comment:
This is done automatically by the mixin, you shouldn't need to define this I don't think. 

Was there a reason to adding this?

Reply:
@Aiky30 Yes you are correct. Thought it was required but tested without it and it's not necessary.